src/main/groovy/me/tatarka/RetrolambdaPluginAndroid.groovy

```diff
@@ -123,7 +123,8 @@ public class RetrolambdaPluginAndroid implements Plugin<Project> {
         }
 
         JavaCompile compileUnitTest = (JavaCompile) project.tasks.find { task -> 
-            task.name.startsWith("compile${variant.capitalize()}UnitTestJava") 
+            task.name.startsWith("compile${variant.capitalize()}UnitTestJava") ||
+            task.name.startsWith("compile${variant.capitalize()}UnitTestJavaWithJavac")
```

Comment:
Yep that would be better. If you change it, can you pull request against the beta branch? I'm about to unbeta that. Otherwise I don't mind doing it.